creator-node/src/premiumContent/premiumContentAccessChecker.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  isCIDForPremiumTrack,
+  isPremiumContentMatch,
+  isRegisteredDiscoveryNode
+} from './helpers'
+import {
+  CheckAccessArgs,
+  CheckAccessResponse,
+  PremiumContentAccessError
+} from './types'
+import { recoverWallet } from '../apiSigning'
+
+export interface AccessChecker {
+  checkPremiumContentAccess(args: CheckAccessArgs): Promise<CheckAccessResponse>
+}
+
+export class PremiumContentAccessChecker implements AccessChecker {
+  async checkPremiumContentAccess({
+    cid,
+    premiumContentHeaders,
+    libs,
+    logger,
+    redis
+  }: CheckAccessArgs): Promise<CheckAccessResponse> {
+    // Only apply premium content middleware logic if file is a premium track file
+    let { trackId, isPremium } = await isCIDForPremiumTrack(cid)
```

Comment:
im a little hesitant on this check bc it does 2 db checks per cid fetch. like just this past day on prod cn1 there were ~2,147,738 cid fetches. this is with cloudflare and nginx cache enabled. is there a way we can cache fetching cid and track data from the db? \ maybe we can do some early exits like checking the blacklist first

Reply:
hmm interesting. good call. should be able to at least reduce the db checks to 1 by performing a join

Reply:
^ changed the 2 queries to be one join query, should hopefully help a bit

Reply:
still a little weary bc similarly, we used to have a select(1) in the health check route, which is a very popular route. this was rly exhausting the db connections. as we're working on rolling users onto other nodes, there are going to be a lot more of these content requests to sync data over. I have a hunch this mw might become a bottleneck. we can chat ab this over a call

Reply:
we can also skip the premium check if the requester is a content node needing data for syncs. would require a "is requester a valid sp" check

Reply:
fwiw testing the query on a prod db snapshot comes back automatically, so we may be fine here

Reply:
still does not confirm that we won't have db connection issues based on the request load that prod will see. let's see how it goes with staging...